Quiet appearance in `SelectInput` component

chloe0592 opened the pull request:
In this PR we added a new 'Quiet' appearance option for the SelectInput component.

Design: [link here](https://www.figma.com/file/hcHKsEwhJ5VXZ4qr5VJNIB/Select-Inputs%3A-Quite-appearance?node-id=101%3A1915&mode=dev).

**isMulti:**
![Screenshot 2023-11-14 at 12 11 04](https://github.com/commercetools/ui-kit/assets/52276952/b3a0d6e2-ec8e-4a10-8e3e-64e166c90219)

**Hover:**
![Screenshot 2023-11-14 at 12 12 24](https://github.com/commercetools/ui-kit/assets/52276952/231df676-0fb6-4ee4-853b-dc74b3890bc9)

FilPob commented:
@chloe0592 thanks! Looks good from my side...  It would be great to mention somewhere as a guideline that using the quite appearance should be paired with using the h.constraint: `auto`

CarlosCortizasCT commented on .changeset/dirty-dragons-flash.md:
```diff
@@ -0,0 +1,7 @@
+'@commercetools-uikit/design-system': minor
+---
+
+Creating a new theme 'Quite' for the `SelectInput` component.
```
Perhaps we want to avoid `theme` word here to prevent confusion.
Maybe we can explain about the new `appearance` property and its available values.

chloe0592 replied:
Thanks Carlos! I agree it's better if we keep the word _appearance_ instead - https://github.com/commercetools/ui-kit/pull/2644/commits/bda3dab4b7ab0e6b50ffd2393d8afd9a61f8db08

CarlosCortizasCT reviewed (commented):
I noticed a little horizontal displacement when the input has an error or a warning in the `quite` appearance.
Users might notice it when entering an invalid value in the input.

Could you maybe take a look if this is something we can adjust? 🙏 

[Components _ Inputs _ SelectInputs - SelectInput ⋅ Storybook.webm](https://github.com/commercetools/ui-kit/assets/97907068/e3f51ba3-be12-4b77-bc91-a39df5f2278f)

kark reviewed (commented):
This is really cool 😎. In addition to what Carlos mentioned about the displacement, I'd like to highlight that we might want to generate a new README for the select input.

PS: Regarding the use of 'quite,' I just wanted to confirm if we intended to use 'quiet' in this context. If it's intentional, please disregard my comment 🙈

FilPob commented:
> This is really cool 😎. In addition to what Carlos mentioned about the displacement, I'd like to highlight that we might want to generate a new README for the select input.
> 
> PS: Regarding the use of 'quite,' I just wanted to confirm if we intended to use 'quiet' in this context. If it's intentional, please disregard my comment 🙈

What would we do without Kacper the english grammar wizard! :D Yes! It should be "quiet" off course. I got it wrong all over Figma as well

chloe0592 commented:
> This is really cool 😎. In addition to what Carlos mentioned about the displacement, I'd like to highlight that we might want to generate a new README for the select input.
> 
> PS: Regarding the use of 'quite,' I just wanted to confirm if we intended to use 'quiet' in this context. If it's intentional, please disregard my comment 🙈

Thank you @kark! 🍫 Good point about README, I'll also explain the usage there (what Filip mentioned)

chloe0592 commented:
> > This is really cool 😎. In addition to what Carlos mentioned about the displacement, I'd like to highlight that we might want to generate a new README for the select input.
> > PS: Regarding the use of 'quite,' I just wanted to confirm if we intended to use 'quiet' in this context. If it's intentional, please disregard my comment 🙈
> 
> What would we do without Kacper the english grammar wizard! :D Yes! It should be "quiet" of course. I got it wrong all over Figma as well

Oh I didn't even think twice about the name. 😆  Great catch @kark! 🦅

chloe0592 commented:
Fixed the typo and renamed '**quite**' to '**quiet**' here - https://github.com/commercetools/ui-kit/pull/2644/commits/4032ff373a169cddfbee5cf08bf06956cb001ad8.

cc/ @kark @FilPob

CarlosCortizasCT reviewed (commented):
Currently when I visit the [preview deployment](https://ui-b3b4no23z-commercetools.vercel.app/?path=/story/components-inputs-selectinputs--selectinput), we've lost the border when in `quite` appearance and having an error.

[Components _ Inputs _ SelectInputs - SelectInput ⋅ Storybook (1).webm](https://github.com/commercetools/ui-kit/assets/97907068/4ff1adc7-67a9-4b27-8ea6-5f95a711df58)

chloe0592 commented:
> Currently when I visit the [preview deployment](https://ui-b3b4no23z-commercetools.vercel.app/?path=/story/components-inputs-selectinputs--selectinput), we've lost the border when in `quite` appearance and having an error.
> 
>  Components._.Inputs._.SelectInputs.-.SelectInput.Storybook.1.webm

Oooh I have no clue, what happened. I already pushed one more commit in the meantime, it looks like it's working alright. Could you also check, please? @CarlosCortizasCT

CarlosCortizasCT commented:
> Oooh I have no clue what happened. I already pushed one more commit in the meantime, it looks like it's working alright. Could you also check, please? @CarlosCortizasCT

Maybe I'm missing something, but I still see the no margin issue.

Using [this preview deployment](https://ui-96xh7qt9j-commercetools.vercel.app/?path=/story/components-inputs-selectinputs--selectinput) which seems to be the last one.

chloe0592 commented on packages/components/inputs/select-utils/src/create-select-styles.ts:
```diff
@@ -115,11 +146,6 @@ const controlStyles = (props: TProps) => (base: TBase, state: TState) => {
       : {}),
 
     '&:focus-within': {
-      boxShadow: (() => {
```
Removing this since it creates unnecessary `3px` green border when focused and in error or warning states. We just need a `2px` border here.

https://github.com/commercetools/ui-kit/assets/52276952/1217d217-7153-4361-8018-1f823b35148d

chloe0592 commented on .changeset/dirty-dragons-flash.md:
```diff
@@ -0,0 +1,7 @@
+'@commercetools-uikit/design-system': minor
+---
+
+Creating a new appearance 'Quiet' for the `SelectInput` component.
```
Done here - [09f15c2](https://github.com/commercetools/ui-kit/pull/2644/commits/09f15c2e45b296d601549e68a5fdca7ecb3b811a). Thanks!

chloe0592 commented:
> > Oooh I have no clue what happened. I already pushed one more commit in the meantime, it looks like it's working alright. Could you also check, please? @CarlosCortizasCT
> 
> Maybe I'm missing something, but I still see the no margin issue.
> 
> Using [this preview deployment](https://ui-96xh7qt9j-commercetools.vercel.app/?path=/story/components-inputs-selectinputs--selectinput) which seems to be the last one.

@FilPob there's just one thing that may be unwanted behaviour. So right now if you don't pick anything in the 'Quiet' input if there's an error state it's very difficult to point that this input is erred. Only the right arrow is red, since we removed the borders for `error` and `warning` states. 
It could become a potential issue if there's for example a form with required inputs and user wants to submit it, but cannot because some of them are erred and they aren't as straightforward since lacking the red borders.
What do you think about it?

![Screenshot 2023-11-16 at 09 23 38](https://github.com/commercetools/ui-kit/assets/52276952/7e75fc60-5023-41c1-aeb7-eb780a96ab79)

FilPob commented:
@chloe0592 I see. An invalidated input should always be paired with an error message, so it wouldn't be so difficult to spot... But that brings me to a question... Is it then necessary to also work on the `Selectfield` component (that can render the error message)? Or is it jut in Storybook, that there is no new knob for the `appearance` in the `Selectfield` component? If yes that we should keep in mind to add that later. So if the input will be paired with an error message, we dont need to change the placeholder font color when invalidated

chloe0592 commented:
> @chloe0592 I see. An invalidated input should always be paired with an error message, so it wouldn't be so difficult to spot... But that brings me to a question... Is it then necessary to also work on the `Selectfield` component (that can render the error message)? Or is it jut in Storybook, that there is no new knob for the `appearance` in the `Selectfield` component? If yes, we should keep in mind to add the knob later. So I think if the input will be paired with an error message, we dont need to change the placeholder font color when invalidated

I'll schedule a quick call for us to discuss this. :)

chloe0592 commented:
@FilPob and @CarlosCortizasCT I pushed the changes we talked about in the meeting - updated the [SelectField](https://github.com/commercetools/ui-kit/pull/2644/commits/1cb22d6329374af7d03df090d9b3899bebe3844d) component and changed the placeholder [font color](https://github.com/commercetools/ui-kit/pull/2644/commits/cb4bf1687b20d95c55026c6b3469da46e3c572af). 
Would you be so kind to give me one more round of your feedback please? Thank you! 🍫

CarlosCortizasCT reviewed (approved):
Awesome! 🙌 

Thanks a lot for all the work in this PR, Klaudija 🙇

kark reviewed (approved):
🎉

Rhotimee commented on packages/components/fields/select-field/README.md:
```diff
@@ -57,6 +57,7 @@ export default Example;
 | `renderError`              | `Function`<br/>[See signature.](#signature-renderError)                                            |          |           | This function can return a message which will be wrapped in an ErrorMessage. It can also return null to show no error.&#xA;<br/>                                                                                                                     ...
 | `isRequired`               | `boolean`                                                                                          |          |           | Indicates if the value is required. Shows an the "required asterisk" if so.                                                                                                                                                                          ...
 | `touched`                  | `union`<br/>Possible values:<br/>`boolean[] , boolean`                                             |          |           | Indicates whether the field was touched. Errors will only be shown when the field was touched.                                                                                                                                                       ...
+| `appearance`               | `union`<br/>Possible values:<br/>`'default' , 'quiet'`                                             |          |           | Indicates the appearance of the input.&#xA;Quite appearence is meant to be used with the `horizontalConstraint="auto"`.                                                                                                                              ...
```
```suggestion
| `appearance`               | `union`<br/>Possible values:<br/>`'default' , 'quiet'`                                             |          |           | Indicates the appearance of the input.&#xA; Quiet appearence is meant to be used with the `horizontalConstraint="auto"`.                                                                                                                                                                                                                                                                  |
```

chloe0592 replied:
@Rhotimee ah thank you for noticing! Fixed here - [ac7dc7a](https://github.com/commercetools/ui-kit/pull/2644/commits/ac7dc7a89da0d9565eba40086de74a23a5286085) ✅

Rhotimee reviewed (commented):
Thanks @chloe0592. Just some spelling corrections.

Rhotimee reviewed (approved):
🙌🏽
